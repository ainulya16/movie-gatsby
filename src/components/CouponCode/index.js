import React from 'react';
import { Card as DefaultCard } from 'antd';
import { Input, Label } from '@aia-digital/ui-library';
import styled from 'styled-components';
import Card from '../Card';

const CouponCodeText = styled.span`
  font-size: 22px;
  font-weight: 900;
`;

const Tag = styled.span`
  background-color: ${({ bgColor }) => bgColor || 'white'};
  border: 2px solid ${({ bgColor }) => bgColor || 'black'};
  color: ${({ textColor }) => textColor || 'black'};
  padding: 5px;
  font-size: 18px;
  font-weight: 900;
`;

const ReferralCodeSection = styled.div`
  margin-top: 10px;
`;

const ReferralCodeTittle = styled(Label)`
  font-weight: bold;
  font-size: 15px;
`;

const CardContainer = styled(Card)`
  margin-top: 20px;
  max-width: 90%;
  background-color: #f5f5f5;
  /* Extra small devices (phones, less than 768px) */
  @media (max-width: 768px) {
    max-width: 100%;
  }
`;

const CouponCode = () => (
  <CardContainer>
    <DefaultCard>
      <CouponCodeText>Coupon Code</CouponCodeText>
      <Tag style={{ marginLeft: '10px' }}>15OFF</Tag>
      <Tag bgColor="#D31145" textColor="white" style={{ marginLeft: '10px' }}>
        {'APPLIED!'}
      </Tag>
    </DefaultCard>
    <ReferralCodeSection>
      <ReferralCodeTittle> Referral Code </ReferralCodeTittle>
      <Input
        style={{ marginTop: '5px', width: '100%' }}
        placeholder="Referral code"
      />
    </ReferralCodeSection>
  </CardContainer>
);

export default CouponCode;
