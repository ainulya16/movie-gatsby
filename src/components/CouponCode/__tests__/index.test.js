import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CouponCode from '../index';

describe('Coupon Code component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CouponCode />);
  });
  it('should be exists', () => {
    expect(wrapper).toBeDefined();
  });
  // it('should contain 1 Card component', () => {
  //   expect(wrapper.find(Card)).toHaveLength(1);
  // });
  it('should match the snapshot', () => {
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
