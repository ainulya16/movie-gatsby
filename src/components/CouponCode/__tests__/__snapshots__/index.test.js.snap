// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon Code component should match the snapshot 1`] = `
.c7 {
  position: relative;
}

.c7 input {
  display: inline-block;
  background-color: #eef0f2;
  width: 100%;
  height: 40px;
  border: none;
  color: #333;
  font-size: 14px;
}

.c7 i {
  position: absolute;
  top: 50%;
}

.c7 .right-icon-input {
  -webkit-transform: translate(-35px,-50%);
  -ms-transform: translate(-35px,-50%);
  transform: translate(-35px,-50%);
}

.c7 .left-icon-input {
  -webkit-transform: translate(10px,-50%);
  -ms-transform: translate(10px,-50%);
  transform: translate(10px,-50%);
}

.c2 {
  font-size: 22px;
  font-weight: 900;
}

.c3 {
  background-color: white;
  border: 2px solid black;
  color: black;
  padding: 5px;
  font-size: 18px;
  font-weight: 900;
}

.c4 {
  background-color: #D31145;
  border: 2px solid #D31145;
  color: white;
  padding: 5px;
  font-size: 18px;
  font-weight: 900;
}

.c5 {
  margin-top: 10px;
}

.c6 {
  font-weight: bold;
  font-size: 15px;
}

.c0 {
  margin-top: 20px;
  max-width: 90%;
  background-color: #f5f5f5;
}

@media (max-width:992px) {
  .c1 {
    padding: 0 40px;
  }
}

@media (max-width:768px) {
  .c0 {
    max-width: 100%;
  }
}

<div
  className="c0 c1"
>
  
  
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <span
        className="c2"
      >
        Coupon Code
      </span>
      <span
        className="c3"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        15OFF
      </span>
      <span
        className="c4"
        style={
          Object {
            "marginLeft": "10px",
          }
        }
      >
        APPLIED!
      </span>
    </div>
  </div>
  <div
    className="c5"
  >
    <label
      className="c6"
    >
       Referral Code 
    </label>
    <div
      className="c7"
      radius={null}
      style={Object {}}
    >
      <input
        placeholder="Referral code"
        style={
          Object {
            "marginTop": "5px",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
</div>
`;
